pipelines/batch/src/main/java/com/google/fhir/analytics/FhirSearchUtil.java

```diff
@@ -334,4 +341,69 @@ void testFhirConnection() {
     query.execute();
     log.info("Validating FHIR connection successful");
   }
+
+  /**
+   * Queries HAPI FHIR MDM for the golden resource ID
+   *
+   * @param fhirId of the resource
+   * @return the golden resource's ID or the same ID if none was found
+   */
+  public String getGoldenResourceFromId(String fhirId, boolean treatPossibleMatchesAsMatches) {
```

Comment:
**question:** This code was likely stolen from the `HTTP` pipeline methods, should we consolidate them to use the same code where possible?